client/session.go

```diff
@@ -172,11 +175,20 @@ func (s *session) Open() error {
 
 	// NB(r): Alloc pools that can take some time in Open, expectation
 	// is already that Open will take some time
-	s.writeOpPool = newWriteOpPool(s.opts.WriteOpPoolSize())
+	writeOpPoolOpts := pool.NewObjectPoolOptions().
+		SetSize(s.opts.WriteOpPoolSize()).
+		SetMetricsScope(s.scope.SubScope("write-op-pool"))
+	s.writeOpPool = newWriteOpPool(writeOpPoolOpts)
 	s.writeOpPool.Init()
-	s.fetchBatchOpPool = newFetchBatchOpPool(s.opts.FetchBatchOpPoolSize(), s.fetchBatchSize)
+	fetchBatchOpPoolOpts := pool.NewObjectPoolOptions().
+		SetSize(s.opts.FetchBatchOpPoolSize()).
+		SetMetricsScope(s.scope.SubScope("fetch-batch-op-pool"))
+	s.fetchBatchOpPool = newFetchBatchOpPool(fetchBatchOpPoolOpts, s.fetchBatchSize)
 	s.fetchBatchOpPool.Init()
-	s.seriesIteratorPool = encoding.NewSeriesIteratorPool(s.opts.SeriesIteratorPoolSize())
+	seriesIteratorPoolOpts := pool.NewObjectPoolOptions().
+		SetSize(s.opts.SeriesIteratorPoolSize()).
+		SetMetricsScope(s.scope.SubScope("series-iterator-pool"))
+	s.seriesIteratorPool = encoding.NewSeriesIteratorPool(seriesIteratorPoolOpts)
 	s.seriesIteratorPool.Init()
 	s.seriesIteratorsPool = encoding.NewMutableSeriesIteratorsPool(s.opts.SeriesIteratorArrayPoolBuckets())
```

Comment:
You can always pass noop scope as the metric scope if you wish.  I figure all or nothing.  We can find a way to opt out of certain pools being instrumented if possible.